### PR TITLE
[pg]: Add QueryResult array possibility if query has multiple statements

### PR DESCRIPTION
**Pull request by fluggo**
This is sure to be unpopular, because it ruins the library's usability, but multi-statement results were added to pg back in v7.0.0:

https://github.com/brianc/node-postgres/commit/28b330c88ebac2ea3ad565f8f2ce4ebe5c9a5579

This means that any query could possibly return an array of results instead of just one, and there's no way for TypeScript to distinguish the two cases. It's completely dynamic. So the user has to cast to one of the two types.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/brianc/node-postgres/commit/28b330c88ebac2ea3ad565f8f2ce4ebe5c9a5579
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Review by aleung — APPROVED** [review]
It's a breaking change. 

Don't know how the version of typing package is generated. Any idea?

**Comment by fluggo** [comment]
> It's a breaking change.
> 
> Don't know how the version of typing package is generated. Any idea?

It's generated from the first line of the source file. I left that line alone, so this becomes a patch change. I left it alone because _ideally_ that line matches the types to a major/minor version of the library itself.

But you're right, this _is_ a breaking change and I kind of want more comment before it's merged.

*MAINTAINERS*: Please don't merge yet. Let's get more time for comments.

**Comment by sandersn** [comment]
I am just a DT maintainer, but my opinion is that this break should wait until a major version of pg, and that people who otherwise need it can just cast manually until then. 8 is the next one I guess?

**Comment by 43081j** [comment]
I was browsing around and saw this.

If you didn't want to diverge versions/introduce a breaking change, you could consider introducing a type parameter which extends the union type and defaults to the non-array.

y'know like `query<T extends (QueryResult|QueryResult[]) = QueryResult>(queryConfig: QueryConfig): Promise<T>;` 🤷‍♂️

**Review by uniqueiniquity — CHANGES_REQUESTED** [review]
Marking as changes requested until the next major version of `pg` is released, at which time this package can also increase its major version.

**Comment by fluggo** [comment]
That's fair, I hadn't thought of that, @43081j. I'll find some time to fix up my patch.

**Pull request closed without merging**
